testutil/pr/track.go

```diff
@@ -0,0 +1,378 @@
+// https://docs.github.com/en/graphql/reference/input-objects#updateprojectv2itemfieldvalueinput
+type UpdateProjectV2ItemFieldValueInput struct {
+	ClientMutationID githubv4.String     `json:"clientMutationId,omitempty"`
+	FieldID          githubv4.ID         `json:"fieldId"`
+	ItemID           githubv4.ID         `json:"itemId"`
+	ProjectID        githubv4.ID         `json:"projectId"`
+	Value            ProjectV2FieldValue `json:"value"`
+}
+
+// AddProjectV2ItemByIdInput is an input for AddProjectV2ItemById.
+// https://docs.github.com/en/graphql/reference/input-objects#addprojectv2itembyidinput
+type AddProjectV2ItemByIdInput struct { //nolint:revive,stylecheck
+	// The ID of the Project to add the item to. (Required.)
+	ProjectID githubv4.ID `json:"projectId"`
+	// The content id of the item (Issue or PullRequest). (Required.)
+	ContentID githubv4.ID `json:"contentId"`
+
+	// A unique identifier for the client performing the mutation. (Optional.)
+	ClientMutationID *githubv4.String `json:"clientMutationId,omitempty"`
+}
+
+// config represents the input fields used in graphql mutations.
+type config struct {
+	projectID       githubv4.ID
+	statusFieldID   githubv4.ID
+	doneOptionID    githubv4.String
+	sizeFieldID     githubv4.ID
+	sprintFieldID   githubv4.ID
+	currIterationID githubv4.ID
+}
+
+// Track tracks a PR without a ticket and adds it to the GitHub board.
+func Track() error {
+	// Ensure only PRs with no associated ticket gets through.
+	unticketed, err := Unticketed()
+	if err != nil {
+		return errors.Wrap(err, "check pr ticket")
+		// log.Fatalf("check pr ticket: %s\n", err.Error())
+	} else if !unticketed {
+		log.Println("ticket exists for this PR")
+		os.Exit(0)
+	}
+
+	ghToken, ok := os.LookupEnv("GH_TOKEN")
+	if !ok {
+		return errors.Wrap(err, "github token not found")
+	}
+
+	src := oauth2.StaticTokenSource(
+		&oauth2.Token{AccessToken: ghToken},
+	)
+
+	httpClient := oauth2.NewClient(context.Background(), src)
+	client := githubv4.NewClient(httpClient)
+
+	// Step 1: Get project data
+	conf, err := getProjectData(client, organization, projectNumber)
+	if err != nil {
+		return errors.Wrap(err, "failed to get project data")
+	}
+
+	// Step 2: Add PR to project board
+	itemID, err := addToProject(client, conf.projectID)
+	if err != nil {
+		return errors.Wrap(err, "failed to add to project")
+	}
+
+	// Step 3: Set size, status and sprint iteration fields
+	err = setFields(client, itemID, conf)
+	if err != nil {
+		return errors.Wrap(err, "failed to set fields")
+	}
+
+	return nil
+}
+
+// getProjectData gets project data given the name of the GitHub organization and the project id.
+func getProjectData(client *githubv4.Client, org string, projectNumber githubv4.Int) (config, error) { //nolint:gocognit
+	variables := map[string]interface{}{
+		"org":    githubv4.String(org),
+		"number": projectNumber,
+	}
+
+	// Note that we query for the first 25 fields.
+	var query struct {
```

Comment:
why is this function local while other JSON types are package level?

Reply:
i refactored all local & package level types to a new file `graphql.go`